### PR TITLE
Upgrade dependencies to Servlet 6.0  and Faces 4.0 (Jakarta EE 10)

### PR DESCRIPTION
**Pull request by tfr42**
This PR upgrades the dependencies from `javax.` to `jakarta.` modules for Servlet API and Faces API.
To fit the new API also Apache commons-fileupload #1625 is migrated to Apache commons-fileupload2 for Servlet 6.0 and a workaround for the incompatible reflection API #1624 is added.

### CODE REVIEW

File: deegree-tests/pom.xml [hunk 1]
```diff
@@ -89,6 +89,13 @@
             </goals>
           </execution>
         </executions>
+        <dependencies>
+          <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.15.1</version>
+          </dependency>
+        </dependencies>
```
Comment (stephanr) [inline deegree-tests/pom.xml:98]: Adding a dependency here is a little unusual.

File: pom.xml [hunk 2]
```diff
@@ -593,38 +593,45 @@
...
       <dependency>
-        <groupId>javax.el</groupId>
-        <artifactId>javax.el-api</artifactId>
-        <version>3.0.0</version>
+        <groupId>jakarta.el</groupId>
+        <artifactId>jakarta.el-api</artifactId>
+        <version>5.0.1</version>
         <scope>provided</scope>
       </dependency>
+      <!--
```
Comment (dstenger) [inline pom.xml:619]: Why is this library commented out instead of being removed?

**Review by stephanr — APPROVED** [review]
Overall, the PR looks fine

File: deegree-tests/pom.xml (hunk 1 above)
Comment (tfr42) [reply deegree-tests/pom.xml:98]: Currently this dependency is required to run the deegree webapp on Jetty container for integration tests. Without the dependency the container fails to start the deegree webapp.
All dependencies need to be reviewed for final release version.

File: pom.xml (hunk 2 above)
Comment (tfr42) [reply pom.xml:619]: The `jakarta.faces:jakarta.faces-api` is included in `org.glassfish:jakarta.faces`. 
All dependencies need to be reviewed for final release version.

**Comment by tfr42** [comment]
For deegree 3.6 the console was migrated to [Faces 4.0](https://jakarta.ee/specifications/faces/4.0/) using [CDI](https://www.cdi-spec.org/).

**Pull request merged**